### PR TITLE
Reorganize source code

### PR DESCRIPTION
**Pull request by olibre**
Lot's of changes, but original spirit is kept :smile: 

* Reorganize source code, directories, files, renaming...
* Change route `/user` -> `/users`
* Replace `requirements.txt` by `Pipfile` (`pip` -> `pipenv`)
* Update `README.md`

### CODE REVIEW

**Comment by cosmic-byte** [comment]
> Lot's of changes, but original spirit is kept 
> 
> * Reorganize source code, directories, files, renaming...
> * Change route `/user` -> `/users`
> * Replace `requirements.txt` by `Pipfile` (`pip` -> `pipenv`)
> * Update `README.md`

Thank you for the PR @olibre. I will review this asap :)

**Comment by matthewliuswims** [comment]
I think @olibre made some nice changes (e.g. using pipenv). @cosmic-byte would you updating https://www.freecodecamp.org/news/structuring-a-flask-restplus-web-service-for-production-builds-c2ec676de563/ to reflect any new changes?

File: README.md [hunk 1]
```diff
@@ -1,39 +1,286 @@
...
+https://github.com/olibre/flask-restplus-boilerplate
 
-    To run test: make tests
+This is a fork from the original Git repository:  
+https://github.com/cosmic-byte/flask-restplus-boilerplate
 
-    To run application: make run
 
-    To run all commands at once : make all
+Motivation
```
Comment (matthewliuswims) [inline README.md]: potentially move this Motvation section from the README and put into the PR comment or commit description? I don't think new-timers would want this context explanation (since they'd have none). Same for "this is a fork..." and "the current example source code" - don't think you'd want these to be actually merged

File: api/main/config.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import os
 
-# uncomment the line below for postgres database url from environment variable
+# uncomment the line below for postgres data url from environment variable
```
Comment (matthewliuswims) [inline api/main/config.py]: nit: why the change? since below it says `os.environ['DATABASE_URL']`

Comment (olibre) [reply api/main/config.py]: Sorry this is a mistake.
I prefer to keep `postgres database`.
I do not know why I did that change. 😳

File: README.md (hunk 1 above)
Comment (olibre) [reply README.md]: Yes I agree with all your remarks.
I will take time to improve/fix my PR in the coming week…

**Comment by cosmic-byte** [comment]
Thank you for your PR. Please if you can make a more granular pull request that improves what we have does not defeat the purpose of this boilerplate, it would be much appreciated.

**Comment by olibre** [comment]
Sorry I do not use Flask since Oct. 2019, I have switched to FastAPI and now, I am currently working on other programming languages. Thank you for maintaining your awesome open source repository :+1:
